brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -198,8 +264,19 @@ public Factory build() {
       for (i = 0; i < keyToField.length; i++) {
         keyToField[i] = keyToFieldList.get(i);
       }
+
+      Customizer[] customizersCopy;
+      if (redactedFieldNames.isEmpty()) {
+        customizersCopy = customizers.toArray(new Customizer[0]);
+      } else {
+        customizersCopy = new Customizer[customizers.size() + 1];
+        for (i = 0; i < customizers.size(); i++) {
+          customizersCopy[i] = customizers.get(i);
+        }
+        customizersCopy[i] = new RedactOnInject(new LinkedHashSet<>(redactedFieldNames));
```

Comment:
if I thought of this earlier, I would have not added the redact flag :P. this dance ensures old config results in the same behavior (you'll notice no tests commented out)